prime_factor/prime_factor.c

```diff
@@ -0,0 +1,43 @@
+#include <stdio.h>
+
+void prime_generator(unsigned int *ar, int *ptr) {
+    // generates array of prime numbers
+    for (int i = 2; i < 60000 && *ptr < 1000; ++i) {
+        int fl = 0;
+        for (int j = 2; j < i; ++j) {
+            if (i % j == 0) {
+                fl = 1;
+                break;
+            }
+        }
+        if (fl == 0) {
+            ar[*ptr] = i;
+            *ptr = *ptr + 1;
+        }
+    }
+}
+
+int prime_factor(int no) {
+    unsigned int primes[1000]; // prime only contains positive values
+    int idx = 0; // last index of the prime number
+    prime_generator(primes, &idx);
+
+    int cnt = 0; // the no. of prime factor of no
+    // prime factors of no
+    printf("Prime factors: ");
+    for (int i = 0; i < idx && primes[i] <= no; ++i) {
+        if (no % primes[i] == 0) {
+            cnt++;
+            printf("%d ", primes[i]);
```

Comment:
`primes[i]` is `unsigned int`, but printed using `%d`. No prints except `main` functions.